package/gluon-offline-ssid/luasrc/usr/bin/gluon-offline-ssid

```diff
@@ -0,0 +1,233 @@
+-- (must be <= switch_timeframe)
+local FIRST = uci:get('offline-ssid', 'settings', 'first') or 5
+
+-- the Offline-SSID will start with this prefix use something short to leave space for the nodename
+-- (no '~' allowed!)
+local PREFIX = uci:get('offline-ssid', 'settings', 'prefix') or 'FF_Offline_'
+
+local DISABLED = '0'
+if uci:get('offline-ssid', 'settings', 'enabled') then 
+  DISABLED ='1'
+end
+
+-- generate the ssid with either 'nodename', 'mac' or to use only the prefix set to 'none'
+local SETTINGS_SUFFIX = uci:get('offline-ssid', 'settings', 'suffix') or 'nodename'
+
+local SUFFIX
+if SETTINGS_SUFFIX == 'nodename' then
+  local pretty_hostname = require 'pretty_hostname'
+  SUFFIX = pretty_hostname.get(uci)
+  -- 32 would be possible as well
+  if ( string.len(SUFFIX) > 30 - string.len(PREFIX) ) then
+    -- calculate the length of the first part of the node identifier in the offline-ssid
+    local HALF = math.floor((28 - string.len(PREFIX) ) / 2)
+    -- jump to this charakter for the last part of the name
+    local SKIP = string.len(SUFFIX) - HALF
+    -- use the first and last part of the nodename for nodes with long name
+    SUFFIX =string.sub(SUFFIX,0,HALF) .. '...' .. string.sub(SUFFIX, SKIP)
+  end
+elseif SETTINGS_SUFFIX == 'mac' then
+  SUFFIX = util.get_wlan_mac(uci, radio, index, 2)
+  if not macaddr then
+    return
+  end
+else
+  -- 'none'
+  local SUFFIX=''
+end
+local OFFLINE_SSID=PREFIX .. SUFFIX
+
+-- get all SSIDs (replace \' with TICX and back to keep a possible tic in an SSID)
+ONLINE_SSIDs = shell_exec('uci show | grep wireless.client_radio[0-9]\.'
+  .. '| grep ssid | awk -F \'=\'  \'{print $2}\'') .. " "  or "~FREIFUNK~"
+-- set a default if for whatever reason ONLINE_SSIDs is NULL
+-- TODO: | sed "s/\\\'/TICX/g" | tr \' \~ | sed "s/TICX/\\\'/g" 
+
+-- temp file to count the offline incidents during switch_timeframe
+local TMP='/tmp/offline-ssid-count'
+local OFF_COUNT='0'
+if not file_exists(TMP) then 
+  local f = io.open(TMP, 'w+')
+  f:write('0')
+else
+  OFF_COUNT = tonumber(util.readfile(TMP))
+end
+
+-- if TQ_LIMIT_ENABLED is true, the offline ssid will only be set if there is no gateway reacheable
+-- upper and lower limit to turn the offline_ssid on and off
+-- in-between these two values the SSID will never be changed to preven it from toggeling every Minute.
+local TQ_LIMIT_ENABLED = uci:get('offline-ssid', 'settings', 'tq_limit_enabled') or '0'
+
+local CHECK
+if ( TQ_LIMIT_ENABLED == 1 ) then
+  --  upper limit, above that the online SSID will be used
+  local TQ_LIMIT_MAX = uci:get('offline-ssid', 'settings', 'tq_limit_max') or '45'
+  --  lower limit, below that the offline SSID will be used
+  local TQ_LIMIT_MIN = uci:get('offline-ssid', 'settings', 'tq_limit_min') or '35'
+  -- grep the connection quality of the currently used gateway
+  local GATEWAY_TQ=shell_exec('batctl gwl | grep -e "^=>" -e "^\*" | awk -F \'[()]\' \'{print $2}\' | tr -d " "')
```

Comment:
Please also write this in Lua. Maybe I've already coded that in Lua. I can look for my script and give it to you tomorrow. Before merging this, we should put all the connection checks into a separate package and let this package depend on it.

Reply:
This is still open, see below: https://github.com/freifunk-gluon/gluon/pull/1649#issuecomment-475932614

Reply:
> we should put all the connection checks into a separate package and let this package depend on it.

#1684